python/src/aac/validate/_validate.py

```diff
@@ -0,0 +1,63 @@
+from contextlib import contextmanager
+from iteration_utilities import flatten
+
+from aac.lang.context_manager import get_active_context
+from aac.parser import parse, ParsedDefinition
+from aac.plugins.plugin_manager import get_validator_plugins
+from aac.plugins.validators import ValidatorResult
+from aac.validate import ValidationResult, ValidationError
+from aac.validate._validate_definition import validate_definition
+
+
+@contextmanager
+def validate_definitions(user_definitions: list[ParsedDefinition]) -> ValidationResult:
```

Comment:
Two things:
1. It looks like we can extract most of this code into a separate function and just pass the different parameter, if needed, for `validate_source` and `validate_definitions`. Or, perhaps, `validate_source` can call `validate_definitions`.
2. I'd prefer naming these context managers something like `validated_source`/`validated_definitions`, etc since they are intended to be used from a `with` statement. `with <verb>` sounds a bit awkward.

Reply:
To point 2. I was thinking along the lines of `with open(file) as fh:`, but I did some googling and apparently there is a PEP for this: https://peps.python.org/pep-0343/#examples

Though, I'll have to do some more reading to understand the implication of the `__enter__` vs `__exit__` disctinction.